gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/PubSubException.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.pubsub;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.gcloud.BaseServiceException;
+import com.google.gcloud.RetryHelper.RetryHelperException;
+import com.google.gcloud.RetryHelper.RetryInterruptedException;
+
+import java.util.Set;
+
+/**
+ * Pub/Sub service exception.
+ *
+ * @see <a href="https://cloud.google.com/pubsub/error-codes">Google Cloud Pub/Sub error codes</a>
+ */
+public class PubSubException extends BaseServiceException {
+
+  private static final long serialVersionUID = 6434989638600001226L;
+  private static final Set<Error> RETRYABLE_ERRORS = ImmutableSet.of(
```

Comment:
1. pongad@ implemented the flag of whether an exception is retryable in https://github.com/googleapis/gax-java/commit/6679c9518d07fdb5e79d1250e50201e3509ca3cd - note the files have moved:

https://github.com/googleapis/gax-java/blob/master/src/main/java/com/google/api/gax/grpc/ExceptionTransformingCallable.java
https://github.com/googleapis/gax-java/blob/master/src/main/java/com/google/api/gax/grpc/ApiException.java

2. The flattened async entry point didn't really get settled. Since you see a use for it in your layer, we can go ahead and add it.